tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/builtinoperations/FreezeAndIsFrozenTest.java

```diff
@@ -590,4 +591,14 @@ public void testFreezeAndIsFrozenNegativeCases() {
                 {"testFrozenStringArrayModification"}
         };
     }
+
+    @Test
+    public void testDeprecatedWarningForIsReadOnly() {
+        CompileResult result = BCompileUtil.compile(
+                "test-src/expressions/builtinoperations/is_readonly_deprecated_warning.bal");
+        int index = 0;
+        validateWarning(result, index++, "usage of construct 'x.isReadOnly(x)' is deprecated", 22, 9);
+        validateWarning(result, index++, "usage of construct 'value:isReadOnly(x)' is deprecated", 24, 17);
```

Comment:
These warnings messages aren't correct.

Using the entire construct is tracked in https://github.com/ballerina-platform/ballerina-lang/issues/24335.

This should probably also be
```
usage of deprecated construct `value:isReadOnly`
```

or


```
`value:isReadOnly` is deprecated
```